src/nfc_reader/poll.c

```diff
@@ -0,0 +1,129 @@
+// Last working compilation:
+// sudo gcc -I/opt/libnfc-1.8.0/include -I/opt/libnfc-1.8.0 -o /opt/FiberPTS/src/nfc_reader/poll /opt/FiberPTS/src/nfc_reader/poll.c -L/opt/libnfc-1.8.0/utils -lnfc
+#include <stdlib.h>
+#include <err.h>
+#include <inttypes.h>
+#include <signal.h>
+#include <stdio.h>
+#include <stddef.h>
+#include <string.h>
+#include <nfc/nfc.h>
+#include <nfc/nfc-types.h>
+#include "utils/nfc-utils.h"
+
+#define MAX_DEVICE_COUNT 16
+
+static nfc_device *pnd = NULL;
+static nfc_context *context;
+
+// TODO: Need to compile this such that it can be called from python
+// TODO: Fix the Makefile so that it not only compiles the linked files correctly,
+// but also correctly cleans up the compiled files 
+// (i.e. removes the .o files except for the libnfc ones)
+
+// Function to handle the stop signal (SIGINT)
+static void stop_polling(int sig)
```

Comment:
What is the stop polling function for? Is this in case of shutdown or power loss?